server/plugin/api_test.go

```diff
@@ -960,3 +961,165 @@ func TestGetUserChannelsForTeam(t *testing.T) {
+				"context": {
+				  "approvalId": 1234,
+				  "organization": "mockOrganization",
+				  "projectName": "mockProjectName",
+				  "requestType": "mockRequestType"
+				}
+			  }`,
+			updatePipelineApprovalRequestStatus:    http.StatusBadRequest,
+			updatePipelineReleaseApprovalPostError: errors.New("failed to update post"),
+			statusCode:                             http.StatusInternalServerError,
+		},
+		{
+			description: "HandlePipelineApproveOrRejectReleaseRequest: failed to approve/reject the request and fetch approval details",
+			body: `{
+				"post_id": "mockPostID",
+				"channel_id": "mockChannelID",
+				"context": {
+				  "approvalId": 1234,
+				  "organization": "mockOrganization",
+				  "projectName": "mockProjectName",
+				  "requestType": "mockRequestType"
+				}
+			  }`,
+			updatePipelineApprovalRequestStatus: http.StatusBadRequest,
+			getApprovalDetailsError:             errors.New("failed to get approval details"),
+			statusCode:                          http.StatusInternalServerError,
+			getApprovalDetailsStatus:            http.StatusInternalServerError,
+		},
+		{
+			description: "HandlePipelineApproveOrRejectReleaseRequest: invalid payload",
+			body: `{
+				"post_id": "mockPostID",
+				"channel_id": "mockChannelID",
+				"context": {
+				  "approvalId": 1234,
+				  "wrong: "mockOrganization",
+				  "projectName": "mockProjectName",
+				  "requestType": "mockRequestType"
+				}
+			  }`,
+			isPayloadInvalid: true,
+			statusCode:       http.StatusInternalServerError,
+		},
+		{
+			description: "HandlePipelineApproveOrRejectReleaseRequest: failed to approve/reject the request due to server error",
+			body: `{
+				"post_id": "mockPostID",
+				"channel_id": "mockChannelID",
+				"context": {
+				  "approvalId": 1234,
+				  "organization": "mockOrganization",
+				  "projectName": "mockProjectName",
+				  "requestType": "mockRequestType"
+				}
+			  }`,
+			updatePipelineApprovalRequestStatus:       http.StatusInternalServerError,
+			statusCode:                                http.StatusInternalServerError,
+			updatePipelineReleaseApprovalRequestError: errors.New("failed to update the pipeline approval request"),
+		},
+	} {
+		t.Run(testCase.description, func(t *testing.T) {
+			mockAPI.On("LogError", testutils.GetMockArgumentsWithType("string", 3)...)
+			mockAPI.On("GetDirectChannel", testutils.GetMockArgumentsWithType("string", 2)...).Return(&model.Channel{}, nil)
+			mockAPI.On("SendEphemeralPost", mock.AnythingOfType("string"), mock.AnythingOfType("*model.Post")).Return(&model.Post{})
+			mockAPI.On("UpdateEphemeralPost", mock.AnythingOfType("string"), mock.AnythingOfType("*model.Post")).Return(nil)
+
+			if !testCase.isPayloadInvalid {
+				mockedClient.EXPECT().UpdatePipelineApprovalRequest(gomock.Any(), "mockOrganization", "mockProjectName", "test-userID", 1234).Return(testCase.updatePipelineApprovalRequestStatus, testCase.updatePipelineReleaseApprovalRequestError)
```

Comment:
the constants we have created in other branches use them here as well

Reply:
Created a ticket for this. I am doing the same thing across multiple PRs. Once all the PRs are merged, I will create a separate PR for such small enhancements.
https://brightscout.atlassian.net/browse/MI-2511?atlOrigin=eyJpIjoiNWFlOTMxODJjYWM4NDgwYWJiNjE2N2MxMDdjZDRkNjAiLCJwIjoiaiJ9